libs/stats/odc/stats/_gm_ls_bitmask.py

```diff
@@ -162,15 +164,17 @@ def reduce(self, xx: xr.Dataset) -> xr.Dataset:
         gm = geomedian_with_mads(xx, **cfg)
         gm = gm.rename(self.renames)
 
-        # Rescaling gm bands between 0-10000 (approx) range
-        # for band in gm.data_vars:
-        #     if band in self.bands:
-        #         gm[band] = scale * sr_scale * gm[band] + offset * sr_scale
-        #         gm[band] = gm[band].round().astype(np.uint16)
-        #
-        # # Rescale edev to sr_scale
-        # gm['emad'] = scale * sr_scale * gm['emad']
-        # gm['emad'] = gm['emad'].round().astype(np.uint16)
+        # rescale gm bands into surface reflectance scale
+        for band in gm.data_vars.keys():
+            if band in self.bands:
+                # gm[band] with low value (ie gm[band] < 7270) end up in negative values during sr scaling
+                # so resetting nodata and negative values back to 0 before rounding to uint16
+                gm[band] = self.scale * self.output_scale * gm[band] + self.offset * self.output_scale
+                gm[band] = gm[band].where(gm[band] > 0, 0)
+                gm[band] = gm[band].round().astype(np.uint16) # still getting some unexpected behaviour
```

Comment:
Are we still getting unexpected results? Or did we resolve all the no-data issues?

Reply:
We resolved an issue by resetting  nodata and negative values back to 0. But we also experiencing slight variation on counts before and after results.